cmd/podman/machine/ssh.go

```diff
@@ -115,25 +107,12 @@ func ssh(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("vm %q is not running", mc.Name)
 	}
 
-	username := sshOpts.Username
-	if username == "" {
-		username = mc.SSH.RemoteUsername
+	if mc.HostUser.Rootful {
+		sshOpts.Username = "root"
+	} else {
+		sshOpts.Username = mc.SSH.RemoteUsername
```

Comment:
Wait a moment, sshOpts.Username is set via the cli `--username` now we always overwrite it.

It should be put in `if sshOpts.Username == "" {` for both branche shere so we never overwrite the cli option, user input must win.

Reply:
Yep, you're right. Thanks for catching that.

Reply:
Should be fixed now.